Src/HALAL/Services/Communication/UART/UART.cpp

```diff
@@ -103,6 +166,34 @@ bool UART::is_busy(uint8_t id){
     return false;
 }
 
+bool UART::set_up_printf(UART::Peripheral& uart){
+		if (printf_ready) {
+			return false;
+		}
+
+		optional<uint8_t> id = UART::inscribe(uart);
+
+		if(!id.has_value()){
```

Comment:
```suggestion
		if(not id){
```
Creo que el has_value es redundante

Reply:
A mi me gusta más poner el has_value pero me da igual.